test/e2e/network_policy.go

```diff
@@ -0,0 +1,502 @@
+		err = framework.WaitForPodSuccessInNamespace(f.Client, podClientB.Name, ns.Name)
+		Expect(err).To(HaveOccurred(), "checking client-b could not communicate with server")
+	})
+
+	It("should enforce policy based on Ports [Feature:NetworkPolicy]", func() {
+		ns := f.Namespace
+		setNamespaceIsolation(f, ns, "DefaultDeny")
+
+		// Create Server with Service
+		By("Creating a simple server.")
+		serverPod, service := createServerPodAndService(f, ns, serverName, []int{listeningPort1, listeningPort2})
+		defer func() {
+			By("Cleaning up the server.")
+			if err := f.Client.Pods(ns.Name).Delete(serverPod.Name, nil); err != nil {
+				framework.Failf("unable to delete pod %v: %v", serverPod.Name, err)
+			}
+		}()
+		defer func() {
+			By("Cleaning up the server's service.")
+			if err := f.Client.Services(ns.Name).Delete(service.Name); err != nil {
+				framework.Failf("unable to delete svc %v: %v", service.Name, err)
+			}
+		}()
+		framework.Logf("Waiting for Server to come up.")
+		err := framework.WaitForPodRunningInNamespace(f.Client, serverPod)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Creating a network policy for the Service which allows traffic only to one port.")
+		policy, err := f.Client.NetworkPolicies(ns.Name).Create(&extensions.NetworkPolicy{
+			ObjectMeta: api.ObjectMeta{
+				Name: fmt.Sprintf("allow-ingress-on-port-%d", listeningPort1),
+			},
+			Spec: extensions.NetworkPolicySpec{
+				// Apply to server
+				PodSelector: unversioned.LabelSelector{
+					MatchLabels: map[string]string{
+						"pod-name": serverPod.Name,
+					},
+				},
+				// Allow traffic only to one port.
+				Ingress: []extensions.NetworkPolicyIngressRule{{
+					Ports: []extensions.NetworkPolicyPort{{
+						Port: &intstr.IntOrString{IntVal: listeningPort1},
+					}},
+				}},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			By("Cleaning up the policy")
+			if err := f.Client.NetworkPolicies(ns.Name).Delete(policy.Name, nil); err != nil {
+				framework.Failf("unable to delete policy %v: %v", policy.Name, err)
+			}
+		}()
+
+		By("Creating client-a that should succesfully connect to the opened port.")
+		podClientA := createClientPod(f, ns, clientAName, service, listeningPort1)
+		defer func() {
+			By("Cleaning up the pod Server A")
+			if err := f.Client.Pods(ns.Name).Delete(podClientA.Name, nil); err != nil {
+				framework.Failf("unable to delete pod %v: %v", podClientA.Name, err)
+			}
+		}()
+
+		framework.Logf("Waiting for client-a to come up.")
+		err = framework.WaitForPodRunningInNamespace(f.Client, podClientA)
+		Expect(err).NotTo(HaveOccurred(), "waiting for client-a to come up.")
+		framework.Logf("Waiting for client-a to complete.")
+		err = framework.WaitForPodSuccessInNamespace(f.Client, podClientA.Name, ns.Name)
+		Expect(err).NotTo(HaveOccurred(), "checking Client-A could communicate with server.")
+
+		By("Creating a client-b that should not succesfully connect to the closed port.")
```

Comment:
"closed" sounds like it means "has had close() called on it". Something like "blocked" or "isolated" would be better. (And "unblocked"/"unisolated" rather than "opened" before)